### ISSUE TITLE
solve: list of equations containing True fails (treated as inequality?)

### ISSUE THREAD

**Issue by cbm755** [sympy/sympy#14632]
Multiple equations in a list are supported and work fine:
```
>>> solve([Eq(x + y, 0),  Eq(2*x + 2*y, 0)], x, y)
{x: -y}
```

But suppose one of the equations reduces to True:
```
>>> solve([Eq(x + y, 0),  Eq(0, 0)], x, y)
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<ipython-input-22-c931741e8ee0> in <module>()
----> 1 solve([Eq(x + y, 0),  Eq(0, 0)], x, y)

/usr/lib/python2.7/site-packages/sympy/solvers/solvers.pyc in solve(f, *symbols, **flags)
    846             f[i] = fi.as_expr()
    847         elif isinstance(fi, (bool, BooleanAtom)) or fi.is_Relational:
--> 848             return reduce_inequalities(f, symbols=symbols)
    849 
    850         # rewrite hyperbolics in terms of exp

/usr/lib/python2.7/site-packages/sympy/solvers/inequalities.pyc in reduce_inequalities(inequalities, symbols)
    699 
    700     # solve system
--> 701     rv = _reduce_inequalities(inequalities, symbols)
    702 
    703     # restore original symbols and return

/usr/lib/python2.7/site-packages/sympy/solvers/inequalities.pyc in _reduce_inequalities(inequalities, symbols)
    620                 raise NotImplementedError(filldedent('''
    621                     inequality has more than one
--> 622                     symbol of interest'''))
    623 
    624         if expr.is_polynomial(gen):

NotImplementedError: 
inequality has more than one symbol of interest
```

If I put the vars in a list, I get a different error:
```
>>> solve([Eq(x + y, 0),  Eq(0, 0)], [x, y])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-10-c89760b4ad5b> in <module>()
----> 1 solve([Eq(x + y, 0),  Eq(0, 0)], [x, y])

/usr/lib/python2.7/site-packages/sympy/solvers/solvers.pyc in solve(f, *symbols, **flags)
    846             f[i] = fi.as_expr()
    847         elif isinstance(fi, (bool, BooleanAtom)) or fi.is_Relational:
--> 848             return reduce_inequalities(f, symbols=symbols)
    849 
    850         # rewrite hyperbolics in terms of exp

/usr/lib/python2.7/site-packages/sympy/solvers/inequalities.pyc in reduce_inequalities(inequalities, symbols)
    669     if not iterable(symbols):
    670         symbols = [symbols]
--> 671     symbols = (set(symbols) or gens) & gens
    672     if any(i.is_real is False for i in symbols):
    673         raise TypeError(filldedent('''

TypeError: unhashable type: 'list'
```

Idea here is that adding True to a list of equations should not change the result (b/c True is always satisfied).  Adding False to the list, I would expect no solutions (`Eq(42, 0)` is never satisfied).

**Comment by asmeurer** [comment]
Related issue https://github.com/sympy/sympy/issues/6924 (except I agree with what you are saying, not with what I said 5 years ago).

**Comment by asmeurer** [comment]
For nonsystems, we have

```
>>> solve(True)
True
>>> solve(False)
False
```

which seems correct enough (anyway, the output type of `solve` is not consistent). 

For solveset:

```
>>> solveset(True)
S.Complexes
>>> solveset(False)
EmptySet()
```

So I think the issue here is just with solving systems.

**Comment by cbm755** [comment]
Might be fixed:
```
>>> solve([Eq(x + y, 0),  Eq(0, 0)], x, y)
{x: -y}
```
Should bisect and make sure there is a test.  And if fixed by something unrelated, let's add this test explicitly.

**Comment by oscarbenjamin** [comment]
This was fixed in #16190.
